multishard_mutation_query.cc

```diff
@@ -742,29 +742,26 @@ future<typename ResultBuilder::result_type> do_query(
         const dht::partition_range_vector& ranges,
         tracing::trace_state_ptr trace_state,
         db::timeout_clock::time_point timeout,
-        ResultBuilder&& result_builder) {
+        ResultBuilder&& result_builder_) {
     auto ctx = seastar::make_shared<read_context>(db, s, cmd, ranges, trace_state, timeout);
 
-    std::exception_ptr ex;
-
-    try {
-        co_await ctx->lookup_readers(timeout);
-
-        auto [last_ckey, result, unconsumed_buffer, compaction_state] = co_await read_page<ResultBuilder>(ctx, s, cmd, ranges, trace_state,
-                std::move(result_builder));
-
-        if (compaction_state->are_limits_reached() || result.is_short_read()) {
-            co_await ctx->save_readers(std::move(unconsumed_buffer), std::move(*compaction_state).detach_state(), std::move(last_ckey));
+    auto result_builder = std::move(result_builder_);
+    auto f = co_await coroutine::as_future(ctx->lookup_readers(timeout).then([&] {
```

Comment:
We need a comment here explaining why we have this weird code. Otherwise somebody might "clean it up" in the future.